src/datasources/sponsor/gelato.sponsor.service.spec.ts

```diff
@@ -24,7 +25,9 @@ describe('GelatoSponsorService', () => {
     },
   });
 
-  const relayService = new GelatoSponsorService(mockConfigService);
+  const mockRelayer = new GelatoRelay();
```

Comment:
Here we are using the actual (3rd party) implementation of the `GelatoRelay` in the tests. While this should work if mocking is set correctly, I have a suggestion that relies less on a real instance being passed to the `GelatoSponsorService`.

What do you think about the following?

```typescript
const gelatoRelay: GelatoRelay = {
  callWithSyncFee: jest.fn(),
  sponsoredCall: jest.fn(),
  sponsoredCallERC2771: jest.fn(),
  getSignatureDataERC2771: jest.fn(),
  sponsoredCallERC2771WithSignature: jest.fn(),
  isNetworkSupported: jest.fn(),
  getSupportedNetworks: jest.fn(),
  isOracleActive: jest.fn(),
  getGelatoOracles: jest.fn(),
  getPaymentTokens: jest.fn(),
  getEstimatedFee: jest.fn(),
  getTaskStatus: jest.fn(),
};

describe('GelatoSponsorService', () => {
  ...
  
  const mockRelayer = jest.mocked(gelatoRelay);
  const relayService = new GelatoSponsorService(mockConfigService, mockRelayer);
  ...
});
```

You would then write the assertions as:

```typescript
expect(mockRelayer.sponsoredCall).toHaveBeenCalledTimes(1);
```

Reply:
Great idea! I've updated it to use the mock.